### PR TITLE
Helm3 support

### PR DESCRIPTION
**Pull request by xxfatumxx**
(In Helm version 3.x.x `helm home` doesn't work, so I've changed location for **auth.REPO** file. Helm version 2.x.x will work as well.

This pull request makes the following changes:

Helm3 support

### CODE REVIEW

File: push.sh [hunk 1]
```diff
@@ -66,9 +66,16 @@ fi
 
 indent() { sed 's/^/  /'; }
 
+declare HELM3_VERSION="helm version | grep "v3\." | wc -l"
```
Comment (dtshepherd) [inline push.sh]: Need to run in client mode only with `-c` or `--client`.  It may also be safer to run `--short` too.

File: push.sh [hunk 2]
```diff
@@ -66,9 +66,16 @@ fi
...
+
 declare REPO=$1
 declare REPO_URL="$(helm repo list | grep "^$REPO" | awk '{print $2}')/"
+
+if [[ $HELM3_VERSION == "1" ]]; then
```
Comment (dtshepherd) [inline push.sh]: Remove the `wc` above and just do a not empty check with `-n`.

**Comment by xxfatumxx** [comment]
Thank you for the tip @dtshepherd Just made additional commit with changes. Please review when you can.

File: push.sh [hunk 3]
```diff
@@ -66,9 +66,16 @@ fi
 
 indent() { sed 's/^/  /'; }
 
+declare HELM3_VERSION="helm --client --short version | grep "v3\.""
```
Comment (dtshepherd) [inline push.sh]: This doesn’t actually run the helm command.  Need to wrap with `$()`

**Review by dtshepherd — CHANGES_REQUESTED** [review]

File: push.sh (hunk 3 above)
Comment (xxfatumxx) [reply push.sh]: Yes, you're right. Didn't check exact var declaration. Sorry. Fixed.

File: push.sh [hunk 4]
```diff
@@ -66,9 +66,16 @@ fi
 
 indent() { sed 's/^/  /'; }
 
+declare HELM3_VERSION="$(helm --client --short version | grep "v3\.")"
```
Comment (dtshepherd) [inline push.sh]: This needs to be `helm version --client`...

Comment (xxfatumxx) [reply push.sh]: I've just checked on both Helm 2 & Helm 3 - options positioning does not affect the output. It's the same. But changed it anyway. Please check when you have a time. Thank you.

Comment (dtshepherd) [reply push.sh]: At least with Helm 2.9.1, it did not work:
```
$ helm --client version
Error: unknown flag: --client
$ helm version --client
Client: &version.Version{SemVer:"v2.9.1", GitCommit:"20adb27c7c5868466912eebdf6664e7390ebe710", GitTreeState:"clean"}
```

**Review by dtshepherd — APPROVED** [review]

**Comment by eshepelyuk** [comment]
@dtshepherd I don't think this PR gonna work

1. reading from `$HOME/.config/helm/auth.$REPO` would not work, because
   - on windows the different folder is used ( in %APPDATA%)
   - one could redefine XDG_* family of variables to be outside HOME
  So, `helm env` should be used instead to detect the location of configs
1. as far as I see helm3 stores repo data in YAML file, called `repository.yaml`, so to reuse auth one need to parse that yaml
1. USERNAME and PASSWORD definately should not be used, since it gives an illusion that smth is read from helm configs, instead USERNAME is taken from default shell variables

**Comment by xxfatumxx** [comment]
@eshepelyuk `helm env` works only in Helm 3 version. But I absolutely agree that this plugin could be run (in container, for example) under user with no definite home dir or initialized HOME variable. I'll fix this and another small bug with PASSWORD variable in next PR.

**Comment by dtshepherd** [comment]
Yeah a buddy of mine gave me a heads up about the XDG yesterday.  We’ll need to rework this to be a bit more generic.  I don’t think the PR made things any worse though?

**Comment by eshepelyuk** [comment]
Nope, nothing worse. Just mentioning that Helm3 needs much more support and rework.

**Comment by eshepelyuk** [comment]
Maybe it worth to separate Helm2 and Helm3 plugins (using branches, for instance) ?

**Pull request merged**